app/src/main/java/com/example/bigowlapp/model/Notification.java

```diff
@@ -6,34 +6,37 @@
 @IgnoreExtraProperties
 public class Notification extends Model {
 
-    private String type;
+    public enum Type {
+        NONE(Notification.class),
+        SUPERVISION_REQUEST(SupervisionRequest.class),
+        SCHEDULE_REQUEST(ScheduleRequest.class);
+
+        public final Class<? extends Notification> typeClass;
+
+        Type(Class<? extends Notification> typeClass) {
+            this.typeClass = typeClass;
+        }
+    }
+
+    private Type type;
     private Timestamp time;
 
     public Notification() {
         super();
+        this.type = Type.NONE;
     }
 
-    public Notification(String type) {
+    public Notification(Type type) {
         super();
         this.type = type;
     }
 
-    public Notification(String uid, String type) {
-        super(uid);
-        this.type = type;
-    }
-
-    public Notification(String uid, String type, Timestamp time) {
-        super(uid);
-        this.type = type;
-        this.time = time;
-    }
-
-    public String getType() {
+    public Type getType() {
         return type;
     }
 
-    public void setType(String type) {
+    // This private setter is necessary for Firebase data to model mapping
+    private void setType(Type type) {
```

Comment:
Why make this private?
Wouldn't it be better to just leave it as public as probably other classes may need it?

Also, this is considered a code smell by Sonarqube.

Reply:
I made it private to prevent the issue that caused the bug in  the first place, the setter being used to change the notifs type.

Ideally I would have simply removed the setter to fit the OOP principle that the type should be read only, but the setter is needed for Firebase model mapping.

Making it private allows firebase mapping to work, while also enforcing the read only nature of this property

Reply:
Alright.

Since it's the cause of the bug, I understand why it should be private.